src/main/scala/org/allenai/scholar/metrics/ErrorAnalysis.scala

```diff
@@ -0,0 +1,41 @@
+package org.allenai.scholar.metrics
+
+import java.time.Year
+
+import org.allenai.scholar._
+import org.allenai.scholar.metrics.PrecisionRecall._
+
+case class ErrorAnalysis(
+  metricName: String,
+  pr: PR,
+  details: Iterable[(String, Example[_])]
+)
+
+object ErrorAnalysis {
+  def computeMetrics[T](
+    truth: Map[String, T],
+    predicted: Map[String, T],
+    metrics: (String, T => Iterable[_])*
+  ) = {
+
+    val examples: Iterable[(String, (String, Example[_]))] =
+      for {
+        (id, trueData) <- truth.toList
```

Comment:
I see. Do we expect to have multiple `extract`s for the same `metric`? The current usage says no. 

Also, `truth` does not need to be a map if used this way. It can be a list of tuples, for example.